### PR TITLE
feat: [net] add "credentials" option to net.request

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
This adds a new option to `net.request()`, "credentials", which more closely matches the behavior of `fetch()` than does "useSessionCookies".

If "credentials" is specified it overrides "useSessionCookies", otherwise the default behavior remains unchanged.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Added a new "credentials" option for `net.request()`.

### CODE REVIEW

**Comment by nornagon** [comment]
NB. this change depends on the current master roll: https://github.com/electron/electron/pull/24984

**Review by deepak1556 — APPROVED** [review]
Api looks better now 👍 

Also `useSessionCookies` was added for https://github.com/electron/electron/issues/8891 where users where only concerned with whether the cookies are completely handled by the store or not. So this change is reasonable for that too.

**Comment by nornagon** [comment]
@MarshallOfSound this is not semver/major because it doesn't break old behavior. it'll be semver/major when we remove the old one.

File: shell/browser/api/electron_api_url_loader.cc [hunk 1]
```diff
@@ -353,6 +376,9 @@ gin::Handle<SimpleURLLoaderWrapper> SimpleURLLoaderWrapper::Create(
   opts.Get("method", &request->method);
   opts.Get("url", &request->url);
   opts.Get("referrer", &request->referrer);
+  bool credentials_specified =
+      opts.Get("credentials", &request->credentials_mode);
+  LOG(INFO) << "credentials_specified = " << credentials_specified;
```
Comment (deepak1556) [inline shell/browser/api/electron_api_url_loader.cc:381]: log can be removed ?

**Review by deepak1556 — APPROVED** [review]
LGTM

**Comment by jkleinsc** [comment]
The API WG approved this at the Sept 21, 2020 meeting.

**Review by jkleinsc — CHANGES_REQUESTED** [review]
This needs a rebase from master to fix the mac arm64 compile issues.

**Comment by nornagon** [comment]
@jkleinsc ping

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**